app/server/lib/DocStorage.ts

```diff
@@ -1462,10 +1462,21 @@ export class DocStorage implements ISQLiteDB, OnDemandStorage {
           ${asLiteral(tableId)} as tableId,
           ${asLiteral(colId)} as colId
         FROM ${quoteIdent(tableId)} AS t, json_each(t.${quoteIdent(colId)}) as a
-        WHERE a.value = ${attId}`);
+        -- json_each is needed because of https://github.com/gristlabs/grist-core/issues/1565 
+        WHERE a.value = ${attId} AND json_valid(t.${quoteIdent(colId)})`);
       }
     }
-    return (await this.all(queries.join(' UNION ALL '))) as any[];
+    try {
+      return (await this.all(queries.join(' UNION ALL '))) as any[];
+    }
+    catch (e) {
+      // We throw an informative error if we fail to process the attachment references, although this shouldn't happen
```

Comment:
Hmm, hard to add a test to exercise this...

Reply:
Yes, please don't make me 😬